dev/integration_tests/display_cutout_rotation/android/app/src/main/java/com/example/display_cutout_rotation/MainActivity.kt

```diff
@@ -0,0 +1,37 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@file:Suppress("PackageName")
+
+package com.example.display_cutout_rotation
+
+import android.os.Build
+import android.os.Bundle
+import android.view.WindowManager
+import androidx.core.view.WindowCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.WindowInsetsControllerCompat
+import io.flutter.embedding.android.FlutterActivity
+
+class MainActivity : FlutterActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // https://developer.android.com/training/system-ui
+        val windowInsetsController = WindowCompat.getInsetsController(window, window.decorView)
+        windowInsetsController.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
+        windowInsetsController.hide(WindowInsetsCompat.Type.systemBars())
+        actionBar?.hide()
+
+        // The default behavior on SDK level 34 and below is for display cutouts to be consumed
+        // before the insets would reach the engine. In order to receive the display cutouts in the
+        // engine, the test app must request that it be allowed to draw its content behind cutouts.
+        // See
+        // https://developer.android.com/reference/android/view/WindowManager.LayoutParams#layoutInDisplayCutoutMode
+        if (Build.VERSION.SDK_INT >= 30) {
```

Comment:
Meta question: Do we plan to run this test on older Android APIs? Are we at this PR? If not, I would consider instead throwing and making sure this test _only_ runs on API >= 30 (or 34, as mentioned) instead of having a condition that we never actually test.

Reply:
Added adb checking for api level.